t/14-grutasks.t

```diff
@@ -317,27 +312,55 @@ is(
     'kept assets for group 1002 accumulated (34 GiB per asset)'
 );
 
-# remove mock tracking data
-unlink $tempdir->child('removed');
-unlink $tempdir->child('deleted');
-
-# now we set the most recent job for asset #1 (99947) to PENDING state,
-# to test protection of assets for PENDING jobs which would otherwise
-# be removed.
-my $job99947            = $schema->resultset('Jobs')->find({id => 99947});
-my $job99947_t_finished = $job99947->t_finished;
-$job99947->update({t_finished => undef});
-
-# Now we run again with size 34GiB. This time asset #1 should again be
-# selected for removal, but reprieved at the last minute due to its
-# association with a PENDING job.
-run_gru_job($t->app, 'limit_assets');
-is(scalar @{mock_removed()}, 1, "only one asset should have been 'removed' at size 34GiB with 99947 pending");
-is(scalar @{mock_deleted()}, 1, "only one asset should have been 'deleted' at size 34GiB with 99947 pending");
-
-# restore job 99947 to DONE state
-$job99947->state(OpenQA::Jobs::Constants::DONE);
-$job99947->update({t_finished => $job99947_t_finished});
+reset_mocked_asset_deletions;
+
+subtest 'assets associated with pending jobs are preserved' => sub {
+    # set the most recent job for asset 1 (which is supposed to be 99947) to PENDING to test protection of
+    # assets for PENDING jobs which would otherwise be removed
+    my $pending_asset      = $assets->find(1);
+    my $pending_asset_name = $pending_asset->name;
+    $pending_asset->update({size => 100000000000, fixed => 0});    # ensure the asset will be deleted
+    my $other_asset_name = 'openSUSE-Factory-staging_e-x86_64-Build87.5011-Media.iso';
+    my $most_reject_job_for_asset_1
+      = $schema->resultset('JobsAssets')->find({asset_id => 1}, {order_by => {-desc => 'job_id'}, rows => 1});
+    $most_reject_job_for_asset_1 = $most_reject_job_for_asset_1->job if $most_reject_job_for_asset_1;
+    is($most_reject_job_for_asset_1->id,
+        99947, "job 99947 is actually the most recent job of asset $pending_asset_name");
+    my $pending_job = $jobs->find(99947);
+    $pending_job->update({state => RUNNING, result => NONE, t_finished => undef});
+
+    # run cleanup again with size 34 GiB; asset 1 should again be selected for removal, but reprieved due to its
+    # association with a PENDING job
+    note('running asset cleanup; pending asset supposed to be preserved');
+    run_gru_job($t->app, 'limit_assets');
+    my ($removed_assets, $deleted_assets) = (mock_removed, mock_deleted);
+    is_deeply($removed_assets, [$other_asset_name],
+        "only other asset $other_asset_name should have been 'removed' at size 34 GiB with 99947 pending")
```

Comment:
to not wrap the lines and also make it easier to understand what can actually be expected, can you somehow make this less ambiguous and shorter? I.e. what does "should have been" mean in this context? Will it be removed or was it removed or not?

Reply:
The 'should have been' is because the removal is mocked. However, it is more confusing indeed so I just make it a 'has been removed'.